.github/workflows/main.yml

```diff
@@ -0,0 +1,113 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - feature/github-actions
+
+jobs:
+  unit_test:
+    name: Unit test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Unit tests
+        run: ./gradlew clean test
+
+  integration_test:
+    name: Integration test
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: beacons-service
+          POSTGRES_DB: beacons
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Integration tests
+        run: ./gradlew clean integrationTest
+        env:
+          SPRING_DATASOURCE_URL: jdbc:postgresql://postgres:5432/beacons
+
+  prettier_check:
+    name: Prettier check
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+      - name: Install npm dependencies
+        run: npm ci
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+      - name: Run prettier
+        run: npm run format:check
+
+  docker_lint:
+    name: Docker lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Docker lint
+        uses: brpaz/hadolint-action@v1.2.1
+
+  docker_build:
+    name: Build and Push Docker image
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Extract branch name
+        shell: bash
+        run: REF=${GITHUB_REF#refs/heads/} && arrREF=(${REF//// }) && echo "##[set-output name=branch;]$(echo ${arrREF[-1]})"
+        id: extract_branch
+      - name: Set up QEMU
```

Comment:
@Ninamma I am not sure if this step is needed here.  It looks like it's required if you're building on different platforms but I would remove this for now and see if it's required for building the image in the step below.

The same can also be done for the webapp pipeline